napari/components/dims.py

```diff
@@ -110,6 +121,26 @@ def mode(self):
         """List of DimsMode: List of DimsMode, one for each dimension."""
         return copy(self._mode)
 
+    @property
+    def axis_labels(self):
+        """List of labels for each axis."""
+        return copy(self._axis_labels)
+
+    @axis_labels.setter
+    def axis_labels(self, labels):
+        if self._axis_labels == labels:
+            return
+
+        if len(labels) != self.ndim:
+            raise ValueError(
+                f"Number of labels doesn't match number of dimensions. Number"
+                f" of given labels was {len(labels)}, number of dimensions is"
+                f" {self.ndim}. Note: If you wish to keep some of the"
+                "dimensions unlabeled, use '' instead."
+            )
+        self._axis_labels = labels
+        self.events.axis_labels()
```

Comment:
do you think this event should emit as
```python
self.events.axis_labels(axis=list(range(self.ndim))))
```
for consistency with the other times it emits? I'm open

Reply:
yeah, this emitter needs an axis kwarg, cause it is connected to
```python
def update_axis_listener(event):
    self.update_axis.emit(event.axis)
```

otherwise, the setter errs:
```python
v = napari.view_image(numpy.random.rand(10,10,10)) 
v.dims.axis_labels = list('xyz')
# throws:  AttributeError: 'Event' object has no attribute 'axis'
```